docs/core/diagnostics/built-in-metrics-runtime.md

```diff
@@ -0,0 +1,242 @@
+- Pinned Object Heap (poh) stores objects allocated using the <xref:System.GC.AllocateArray%2A?displayProperty=nameWithType> API when the `pinned` parameter is true.
+
+Both of these special generations are collected during `gen2` GC collections. For more information about the .NET GC see the [.NET Garbage collection guide](/standard/garbage-collection/fundamentals.md).
+
+Available starting in: .NET 9.0.
+
+##### Metric: `dotnet.gc.last_collection.heap.fragmentation.size`
+
+| Name     | Instrument Type | Unit (UCUM) | Description    |
+| -------- | --------------- | ----------- | -------------- |
+| `dotnet.gc.last_collection.heap.fragmentation.size` | UpDownCounter | `By` | The heap fragmentation, as observed during the latest garbage collection. |
+
+This metric reports the same values as calling <xref:System.GCGenerationInfo.FragmentationAfterBytes?displayProperty=nameWithType>.
+
+When .NET objects are allocated, initially they tend to be layed out contiguously in memory. However if some of those objects are later collected by the GC this creates gaps of unused memory between the live objects that remain. These gaps represent the portion of the GC heap that is not currently being used to store objects, often called "fragmentation."  The GC can reuse the fragmentation bytes in the future for new object allocations if the object size is small enough to fit in one of the gaps. The GC can also perform a special compacting garbage collection that moves remaining live objects next to each other as long as the objects haven't been pinned in place.
+
+For more information about how the .NET GC works, analyzing GC performance, and what role fragmentation plays, see [.NET memory performance analysis](https://github.com/Maoni0/mem-doc/blob/master/doc/.NETMemoryPerformanceAnalysis.md).
+
+Available starting in: .NET 9.0.
+
+##### Metric: `dotnet.gc.pause.time`
+
+| Name     | Instrument Type | Unit (UCUM) | Description    |
+| -------- | --------------- | ----------- | -------------- |
+| `dotnet.gc.pause.time` | Counter | `s` | The total amount of time paused in GC since the process has started. |
+
+This metric reports the same values as calling <xref:System.GC.GetTotalPauseDuration?displayProperty=nameWithType>.
+
+Each time the .NET GC does a collection it needs to briefly pause all threads running managed code to determine which objects are still referenced. This metric reports the sum of all these pause times since the process began. This can be used to determine what fraction of time threads spend paused for GC vs. the time they are able to run managed code.
+
+Available starting in: .NET 9.0.
+
+##### Metric: `dotnet.jit.compiled_il.size`
+
+| Name     | Instrument Type | Unit (UCUM) | Description    |
+| -------- | --------------- | ----------- | -------------- |
+| `dotnet.jit.compiled_il.size` | Counter | `By` | Count of bytes of intermediate language that have been compiled since the process has started. |
```

Comment:
```suggestion
| `dotnet.jit.compiled_il.size` | Counter | `By` | Count of bytes of intermediate language that have been compiled since the process started. |
```